spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -338,6 +338,9 @@ protected void doStop() {
 	protected void actualStart() throws Exception {
 		this.aborted = false;
 		this.hasStopped = false;
+		if (getPrefetchCount() < this.messagesPerAck) {
+			setPrefetchCount(this.messagesPerAck);
```

Comment:
Yeah - most people will have prefetch "slightly" higher than mPA - so a message is immediately available after the ack is sent.